pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,399 @@
+	}
+
+	if cfg.DryRun && !cfg.Evacuate {
+		errList = append(errList, errors.New("--dry-run is only applicable for --evacuate"))
+	}
+
+	if cfg.PodSelector.Provided() {
+		if _, err := labels.Parse(cfg.PodSelector.Value()); err != nil {
+			errList = append(errList, errors.New("--pod-selector=<pod_selector> must be a valid label selector"))
+		}
+	}
+	return errList
+}
+
+// RunManageNode contains all the necessary functionality for the openshift cli manage-node command
+func (cfg *NodeConfig) RunManageNode() error {
+	nameArgs := []string{"nodes"}
+	if len(cfg.NodeNames) != 0 {
+		nameArgs = append(nameArgs, cfg.NodeNames...)
+	}
+
+	r := resource.NewBuilder(cfg.Mapper, cfg.Typer, resource.ClientMapperFunc(cfg.RESTClientFactory)).
+		ContinueOnError().
+		NamespaceParam(cfg.DefaultNamespace).
+		SelectorParam(cfg.Selector.Value()).
+		ResourceTypeOrNameArgs(true, nameArgs...).
+		Flatten().
+		Do()
+	if r.Err() != nil {
+		return r.Err()
+	}
+
+	var err error
+	errList := []error{}
+	nodeCount := 0
+	ignoreHeaders := false
+	err = r.Visit(func(info *resource.Info) error {
+		nodeCount++
+		node, ok := info.Object.(*kapi.Node)
+		if !ok {
+			err = fmt.Errorf("cannot convert input to Node: ", reflect.TypeOf(info.Object))
+			errList = append(errList, err)
+			return nil
+		}
+
+		switch cfg.Operations[0] {
+		case "schedulable":
+			err = cfg.RunSchedulable(node, ignoreHeaders)
+			ignoreHeaders = true
+		case "evacuate":
+			if cfg.DryRun {
+				err = cfg.RunListPods(node)
+			} else if cfg.Evacuate {
+				err = cfg.RunEvacuate(node)
+			}
+		case "list-pods":
+			if cfg.ListPods {
+				err = cfg.RunListPods(node)
+			}
+		}
+
+		if err != nil {
+			errList = append(errList, err)
+			return nil
+		}
+		return nil
+	})
+	if len(errList) != 0 {
+		return fmt.Errorf("%v", errList)
+	}
+	if nodeCount == 0 {
+		if len(cfg.NodeNames) > 0 {
+			return fmt.Errorf("Nodes %v not found", strings.Join(cfg.NodeNames, ","))
+		} else {
+			return fmt.Errorf("No nodes found")
+		}
+	}
+	return nil
+}
+
+func (cfg *NodeConfig) RunSchedulable(node *kapi.Node, ignoreHeaders bool) error {
+	if cfg.Schedulable {
+		if node.Spec.Unschedulable {
+			node.Spec.Unschedulable = false
+		}
+	} else {
+		if !node.Spec.Unschedulable {
+			node.Spec.Unschedulable = true
+		}
+	}
+
+	updatedNode, err := cfg.Kclient.Nodes().Update(node)
+	if err != nil {
+		return err
+	}
+
+	printerWithHeaders, printerNoHeaders, err := cfg.getPrinters("", updatedNode)
+	if err != nil {
+		return err
+	}
+	if ignoreHeaders {
+		printerNoHeaders.PrintObj(updatedNode, cfg.Writer)
+	} else {
+		printerWithHeaders.PrintObj(updatedNode, cfg.Writer)
+	}
+	return nil
+}
+
+func (cfg *NodeConfig) RunEvacuate(node *kapi.Node) error {
+	if !node.Spec.Unschedulable {
+		return fmt.Errorf("Node '%s' must be unschedulable to perform evacuation.\nYou can mark the node unschedulable with openshift admin manage-node %s --schedulable=false", node.ObjectMeta.Name, node.ObjectMeta.Name)
```

Comment:
This was discussed before. If we automatically mark the node unschedulable and perform the evaucate op, user may stop this operation (ctrl-c, that can be handled) or there can be network error(we can't handle this). In this case, node will be unschedulable forever until user changes. Making the user to set unschedulable explicitly will avoid these cases. User can decide when to mark the node schedulable/unschedulable. Aso the error message returns the exact cmd to run.